### DISCUSSION TITLE
How can I pass a parameter when I toggle modal using JavaScript

### DISCUSSION THREAD

**Discussion by FoundAdd** [twbs/bootstrap#34316] [Q&A] [answered]
For example, when I toggle modal by $('#modal').modal('toggle'), how do I attach a parameter to tell modal the source elem

`$('#record-modal').modal('toggle')`

**Comment by GeoSot** [comment, marked as answer]
you can try `$('#record-modal').modal('toggle', my-source-element)`

I think it works for now, but I wouldn't suggest it. 
Instead I would suggest you to use  Modal.getOrCreateInstance('#record-modal').toggle(my-source-element)

PS: `getOrCreateInstance` method will be available in 5.0.2(will be released soon)

**Reply by FoundAdd** [reply]
Thanks for the answer, the first method is indeed effective.I'm looking forward to 5.0.2

`$('#record-modal').modal('toggle', 'edit')`

`'#record-modal').on('show.bs.modal', function (event) {
	var source = event.relatedTarget
	console.log(source)
})`

![image](https://user-images.githubusercontent.com/58539321/122770501-16aae700-d2d8-11eb-9a1b-70f649aa7cf1.png)
